pipelines/build/common/config_regeneration.groovy

```diff
@@ -301,55 +301,50 @@ class Regeneration implements Serializable {
+            context.stage("Check $javaVersion pipeline status") {
 
-                    // Parse api response to only extract the relevant pipeline
-                    getPipelines.jobs.name.each{ pipeline ->
-                        if (pipeline.contains("pipeline") && pipeline.contains(versionNumbers[0])) {
-                            Integer sleepTime = 900
-                            Boolean inProgress = true
+                // Get all pipelines
+                def getPipelines = queryAPI("${jenkinsBuildRoot}/api/json?tree=jobs[name]&pretty=true&depth1")
 
-                            while (inProgress) {
-                                // Check if pipeline is in progress using api
-                                context.println "[INFO] Checking if ${pipeline} is running..." //i.e. openjdk8-pipeline
+                // Parse api response to only extract the relevant pipeline
+                getPipelines.jobs.name.each{ pipeline ->
+                    if (pipeline.contains("pipeline") && pipeline.contains(versionNumbers[0])) {
+                        Integer sleepTime = 900
+                        Boolean inProgress = true
 
-                                def pipelineInProgress = queryAPI("${jenkinsBuildRoot}/job/${pipeline}/lastBuild/api/json?pretty=true&depth1")
+                        while (inProgress) {
+                            // Check if pipeline is in progress using api
+                            context.println "[INFO] Checking if ${pipeline} is running..." //i.e. openjdk8-pipeline
 
-                                // If query fails, check to see if the pipeline been run before
-                                if (pipelineInProgress == null) {
-                                    def getPipelineBuilds = queryAPI("${jenkinsBuildRoot}/job/${pipeline}/api/json?pretty=true&depth1")
+                            def pipelineInProgress = queryAPI("${jenkinsBuildRoot}/job/${pipeline}/lastBuild/api/json?pretty=true&depth1")
 
-                                    if (getPipelineBuilds.builds == []) {
-                                        context.println "[SUCCESS] ${pipeline} has not been run before. Running regeneration job..."
-                                        inProgress = false
-                                    }
+                            // If query fails, check to see if the pipeline been run before
+                            if (pipelineInProgress == null) {
+                                def getPipelineBuilds = queryAPI("${jenkinsBuildRoot}/job/${pipeline}/api/json?pretty=true&depth1")
 
-                                } else {
-                                    inProgress = pipelineInProgress.building as Boolean
+                                if (getPipelineBuilds.builds == []) {
+                                    context.println "[SUCCESS] ${pipeline} has not been run before. Running regeneration job..."
```

Comment:
Is this a case the pipeline has never ever been run before or is there some other time window context here?

Reply:
> Is this a case the pipeline has never ever been run before

That one 😄. The following line fails if the pipeline has never been run as the `building` attribute is not populated
```groovy
Line 331    inProgress = pipelineInProgress.building as Boolean
```